src/builder.rs

```diff
@@ -110,23 +116,24 @@ impl ActionInfo {
     }
 
     /// Returns the value sum for this action.
-    fn value_sum(&self) -> Result<ValueSum, value::OverflowError> {
-        self.spend.note.value() - self.output.value
+    fn value_sum<V: ValueSum>(&self) -> Option<V> {
+        V::sub(self.spend.note.value(), self.output.value)
     }
 
     /// Builds the action.
     ///
     /// Defined in [Zcash Protocol Spec § 4.7.3: Sending Notes (Orchard)][orchardsend].
     ///
     /// [orchardsend]: https://zips.z.cash/protocol/nu5.pdf#orchardsend
-    fn build(
+    fn build<V: ValueSum>(
         self,
         mut rng: impl RngCore,
+        _v: PhantomData<V>, //the compiler needs a hint
```

Comment:
Instead of passing this as a parameter, use [turbofish](https://turbo.fish/) notation at the callsite: `a.build::<V>(rng)`